collectors/ebpf.plugin/ebpf_softirq.c

```diff
@@ -0,0 +1,274 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#include "ebpf.h"
+#include "ebpf_softirq.h"
+
+struct config softirq_config = { .first_section = NULL,
+    .last_section = NULL,
+    .mutex = NETDATA_MUTEX_INITIALIZER,
+    .index = { .avl_tree = { .root = NULL, .compar = appconfig_section_compare },
+        .rwlock = AVL_LOCK_INITIALIZER } };
+
+#define SOFTIRQ_MAP_LATENCY 0
+static ebpf_local_maps_t softirq_maps[] = {
+    {
+        .name = "tbl_softirq",
+        .internal_input = NETDATA_SOFTIRQ_MAX_IRQS,
+        .user_input = 0,
+        .type = NETDATA_EBPF_MAP_STATIC,
+        .map_fd = ND_EBPF_MAP_FD_NOT_INITIALIZED
+    },
+    /* end */
+    {
+        .name = NULL,
+        .internal_input = 0,
+        .user_input = 0,
+        .type = NETDATA_EBPF_MAP_CONTROLLER,
+        .map_fd = ND_EBPF_MAP_FD_NOT_INITIALIZED
+    }
+};
+
+static ebpf_data_t softirq_data;
+
+#define SOFTIRQ_TP_CLASS_IRQ "irq"
+static ebpf_tracepoint_t softirq_tracepoints[] = {
+    {.enabled = false, .class = SOFTIRQ_TP_CLASS_IRQ, .event = "softirq_entry"},
+    {.enabled = false, .class = SOFTIRQ_TP_CLASS_IRQ, .event = "softirq_exit"},
+    /* end */
+    {.enabled = false, .class = NULL, .event = NULL}
+};
+
+// these must be in the order defined by the kernel.
```

Comment:
Considering that kernel is keeping this list relatively stable, I suggest to add this [link](https://elixir.bootlin.com/linux/v5.12.19/source/include/trace/events/irq.h#L13) for the comment.

Reply:
Addressed in https://github.com/netdata/netdata/pull/11445/commits/ba9cbddce1ebb91fccbc059a8fbdd7cc0a386424